### PR TITLE
enable master kubelet in gce-default to match gce-test

### LINKED ISSUE #15803: Enable REGISTER_MASTER_KUBELET by default
To enable this feature by default for OSS k8s, we need to fix a bunch of e2e tests. File this issue for place holder.

### PR DESCRIPTION
**Pull request by mikedanese**
Fixes #15803

### CODE REVIEW

**Comment by wojtek-t** [comment]
@gmarek - FYI

**Comment by dchen1107** [comment]
LGTM

**Comment by gmarek** [comment]
It's a bold move - don't we want to let it cook a bit?

**Comment by mikedanese** [comment]
@gmarek this was enabled for all of v1.1. That's seems sufficient to me.

**Comment by gmarek** [comment]
@mikedanese - the point it, that it was not. It was enabled in production only - we haven't run tests on it.

**Comment by mikedanese** [comment]
ok, this flag that is enabled in this PR was enabled for all v1.1 builds and most v1.2-alpha builds. After much deliberation, we made the decision to go to v1.1 with this flag enabled and without the test flag enabled.  There is no new evidence that would lead me to believe that we made the wrong call and that we should revert that decision.

**Comment by mikedanese** [comment]
cc @dchen1107 @roberthbailey

**Comment by dchen1107** [comment]
Yes, @mikedanese refreshed my memory on this. We made decision for 1.1 release to have this on default for production. If we disabled this, it is a usability regression for the users.

**Comment by roberthbailey** [comment]
@mikedanese I'm a bit confused. If I look at the tip of the release-1.1 branch, I see the default for `REGISTER_MASTER` set to [false](https://github.com/kubernetes/kubernetes/blob/release-1.1/cluster/gce/config-default.sh#L28). So the flag existed in v1.1 but it wasn't enabled by default.

**Comment by mikedanese** [comment]
I am also confused. I completely miss remembered that. Here is the PR https://github.com/kubernetes/kubernetes/pull/15789.

**Comment by roberthbailey** [comment]
@dchen1107 rolled that back in https://github.com/kubernetes/kubernetes/pull/21350 since it was never enabled for e2e tests. Also note that we never cherry picked it into the 1.1 release branch.

**Comment by gmarek** [comment]
This is exactly why I was kind of surprised by making this a cherry-pick candidate.

**Comment by roberthbailey** [comment]
@gmarek -- #18338 was merged into the release branch (as part of #22710) so right now on the release branch config-default and config-test are inconsistent. We should either revert config-test back for the 1.2 release or cherry pick this PR.

**Comment by dchen1107** [comment]
I personally want to cherry pick this PR. It is very useful to watch master component pods' status. But not sure how easy to make this compatible with GKE's configuration.

**Comment by gmarek** [comment]
TL;DR - I think we should cherry-pick this PR, but we should have it mind that this is not a zero-risk thing.

I agree with @dchen1107 that this is an extremely useful thing, and I'd love to cherry pick this into v1.2. The question is if we're confident enough that it's not breaking anything. Tests are generally passing except kubelet-perf for 100 pods, but that's a test's problem - fix is on the way #22743 (if we cherry pick this PR, we should cherry pick that fix as well). So I'm not really worried about short-term stuff, but I don't know about long term thing (not that our soak tests were passing ever) - I don't think there's a high risk, but there's a risk.

**Comment by mikedanese** [comment]
I agree

**Comment by roberthbailey** [comment]
+1 to cherry pick

**Comment by eparis** [comment]
This PR was sucessfully cherry picked in PR #22855
please verify that the release-1.2 branch contains these changes as you would expect and contact @eparis if there appear to be problems.

**Comment by roberthbailey** [comment]
Looks correct in the release branch; thanks @eparis!
